console/src/tasks.rs

```diff
@@ -127,12 +164,23 @@ impl State {
             let fields = task
                 .fields
                 .iter()
-                .fold(String::new(), |mut res, f| {
-                    write!(&mut res, "{} ", f).unwrap();
-                    res
+                .filter_map(|f| {
+                    let field_name = f.name.as_ref().expect("no name");
+                    let name = match field_name {
+                        proto::field::Name::StrName(n) => Some(n.clone()),
+                        proto::field::Name::NameIdx(idx) => {
+                            let meta_id = f.metadata_id.as_ref().expect("no id");
+                            metas
+                                .get(&meta_id.id)
+                                .and_then(|meta| meta.field_names.get(*idx as usize))
+                                .cloned()
```

Comment:
it might be worth representing the metadata's field names as `Arc<str>`s, so that they are cheap to clone --- once the subscriber populates the metadata field names, we can expect most spans to have field names stored as indices rather than strings, so we could significantly reduce the number of strings the console allocates if we only stored a single copy of those strings in the metadata, and cloned an `Arc` into the task's fields.

it would be fine to do this in a follow-up PR, though --- this is fine for now.